src/dotnet/install.sh

```diff
@@ -53,440 +22,145 @@ err() {
-    apt-get install -yq ${DOTNET_PACKAGE}${APT_DOTNET_VERSION}
-    if [ "$?" != 0 ]; then
-        echo "Failed to complete apt install of ${DOTNET_PACKAGE}${TARGET_DOTNET_VERSION}"
-        return 1
-    fi
-
-    # Add symlink for current
-    CURRENT_DIR="${TARGET_DOTNET_ROOT}/current"
-    if [[ -d "${CURRENT_DIR}" ]]; then
-        rm -rf "${CURRENT_DIR}"
-    fi
-    mkdir -p "${TARGET_DOTNET_ROOT}"
-    local dotnet_installed_version="$(dotnet --version)"
-    # See if its the distro version
-    if [[ "$(dotnet --info)" == *"Base Path:   /usr/lib/dotnet/${DOTNET_SDK_OR_RUNTIME}/${dotnet_installed_version}"* ]]; then
-        ln -s "/usr/lib/dotnet" "${CURRENT_DIR}"
+fetch_latest_sdk_version() {
+    local latest_STS=$(fetch_latest_sdk_version_in_channel "STS")
+    local latest_LTS=$(fetch_latest_sdk_version_in_channel "LTS")
+    if [[ "$latest_STS" > "$latest_LTS" ]]; then
+        echo "$latest_STS"
     else
-        # Location used by MS repo versions
-        ln -s "/usr/share/dotnet" "${CURRENT_DIR}" 
+        echo "$latest_LTS"
     fi
 }
 
-# Find and extract .NET binary download details based on user-requested version
-# args:
-# sdk_or_runtime $1
-# dotnet_version_to_download $2
-# exports:
-# DOTNET_DOWNLOAD_URL
-# DOTNET_DOWNLOAD_HASH
-# DOTNET_DOWNLOAD_NAME
-get_full_version_details() {
-    local sdk_or_runtime="$1"
-    local architecture
-    local dotnet_channel_version
-    local dotnet_releases_url
-    local dotnet_releases_json
-    local dotnet_latest_version
-    local dotnet_download_details
-
-    export DOTNET_DOWNLOAD_VERSION="$2"
-    export DOTNET_DOWNLOAD_URL
-    export DOTNET_DOWNLOAD_HASH
-    export DOTNET_DOWNLOAD_NAME
-
-    # Set architecture variable to current user's architecture (x64 or ARM64).
-    architecture="$(get_architecture_name_for_target_os)"
-
-    # Set VERSION to empty string to ensure jq includes all .NET versions in reverse sort below 
-    if [ "${DOTNET_DOWNLOAD_VERSION}" = "latest" ]; then
-        DOTNET_DOWNLOAD_VERSION=""
-    fi
-
-    dotnet_patchless_version="$(echo "${DOTNET_DOWNLOAD_VERSION}" | cut -d "." --field=1,2)"
-
-    set +e
-    dotnet_channel_version="$(curl -s "${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/releases-index.json" | jq -r --arg channel_version "${dotnet_patchless_version}" '[."releases-index"[]] | sort_by(."channel-version") | reverse | map( select(."channel-version" | startswith($channel_version))) | first | ."channel-version"')"
-    set -e
-
-    # Construct the releases URL using the official channel-version if one was found.  Otherwise make a best-effort using the user input.
-    if [ -n "${dotnet_channel_version}" ] && [ "${dotnet_channel_version}" != "null" ]; then
-        dotnet_releases_url="${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/${dotnet_channel_version}/releases.json"
+# Installs a version of .NET using the DOTNET_INSTALLER_SCRIPT
+install_version() {
+    local version="$1"
+    local channel="STS"
```

Comment:
It seems like this default value is never used. Consider setting it to `channel=""` to make that more clear

Reply:
Very true, fixed.